frontend/src/components/orgs-list.ts

```diff
@@ -110,63 +225,204 @@ export class OrgsList extends LiteElement {
       this.dispatchEvent(
         new CustomEvent("update-quotas", { detail: this.currOrg }),
       );
+
+      void this.orgQuotaDialog?.hide();
     }
-    this.currOrg = null;
   }
 
-  private showQuotas(org: OrgData) {
-    const stop = (e: Event) => {
-      e.preventDefault();
-      e.stopPropagation();
-      this.currOrg = org;
-      return false;
-    };
+  private async onSubmitReadOnly(e: SubmitEvent) {
+    e.preventDefault();
+
+    if (!this.currOrg) return;
+
+    const formEl = e.target as HTMLFormElement;
+    if (!(await this.checkFormValidity(formEl))) return;
+
+    const { readOnlyReason } = serialize(formEl) as { readOnlyReason: string };
 
-    return stop;
+    await this.updateReadOnly(this.currOrg, {
+      readOnly: true,
+      readOnlyReason: readOnlyReason,
+    });
+
+    void this.orgReadOnlyDialog?.hide();
   }
 
-  private readonly renderOrg = (defaultOrg?: UserOrg) => (org: OrgData) => {
+  private async updateReadOnly(
+    org: OrgData,
+    params: Pick<OrgData, "readOnly" | "readOnlyReason">,
+  ) {
+    try {
+      await this.api.fetch(`/orgs/${org.id}/read-only`, this.authState!, {
+        method: "POST",
+        body: JSON.stringify(params),
+      });
+
+      this.orgList = this.orgList?.map((o) => {
+        if (o.id === org.id) {
+          return {
+            ...o,
+            ...params,
+          };
+        }
+        return o;
+      });
+
+      this.notify.toast({
+        message: params.readOnly
+          ? msg(str`Org "${org.name}" is read-only.`)
+          : msg(str`Org "${org.name}" is no longer read-only.`),
+        variant: "success",
+        icon: "check2-circle",
+      });
+    } catch (e) {
+      console.debug(e);
+
+      this.notify.toast({
+        message: msg(
+          "Sorry, couldn't update org read-only state at this time.",
+        ),
+        variant: "danger",
+        icon: "exclamation-octagon",
+      });
+    }
+  }
+
+  private readonly renderOrg = (org: OrgData) => {
     if (!this.userInfo) return;
 
     // There shouldn't really be a case where an org is in the org list but
     // not in user info, but disable clicking into the org just in case
     const isUserOrg = this.userInfo.orgs.some(({ id }) => id === org.id);
 
-    let defaultLabel: TemplateResult | undefined;
+    const memberCount = Object.keys(org.users || {}).length;
+
+    const none = html`
+      <sl-icon
+        name="slash"
+        class="text-base text-neutral-400"
+        label=${msg("None")}
+      ></sl-icon>
+    `;
+
+    let status = {
+      icon: html`<sl-icon
+        class="text-base text-success"
+        name="check-circle"
```

Comment:
We don't use the non filled variant for statuses anymore with the exception of `slash-circle` to give it a more unique shape from the rest.

```suggestion
        name="check-circle-fill"
```